### PR TITLE
Fixing a Corner Case Bug in Compose when transforms=[]

### PR DESCRIPTION
**Pull request by immortalCO**
When `transforms` is `[]`, `_set_keys()` will NOT be adding `additional_targets` into `self._available_keys` (as it is just going through all transforms and processors, which is nothing done when there is no transform or processor), and causes runtime errors if passing those in the usage. 
Adding the L151 will fix this corner case.

## Summary by Sourcery

This pull request addresses a corner case bug in the Compose class by ensuring that `additional_targets` are added to `_available_keys` even when `transforms` is an empty list, thereby preventing potential runtime errors.

* **Bug Fixes**:
    - Fixed a corner case bug in the Compose class where `additional_targets` were not being added to `_available_keys` when `transforms` is an empty list, preventing runtime errors.

### CODE REVIEW

File: albumentations/core/composition.py [hunk 1]
```diff
@@ -148,6 +148,7 @@ def add_targets(self, additional_targets: Optional[Dict[str, str]]) -> None:
 
     def _set_keys(self) -> None:
         """Set _available_keys"""
+        self._available_keys.update(self._additional_targets.keys())
```
[the remark this answers is not in the document]
Comment (immortalCO) [reply albumentations/core/composition.py:151]: In L70, `self._additional_targets: Dict[str, str] = {}` which is already not `None`. There are no other assignments to it, so it is guaranteed not to be `None`.

**Review by ternaus — CHANGES_REQUESTED** [review]
Test is needed, so that should be sure that this bug will not happen in the future.

**Comment by immortalCO** [comment]
> Test is needed, so that should be sure that this bug will not happen in the future.

Added a test case for this corner case as `test_compose_additional_targets_in_available_keys()` in `tests/test_core.py`.

**Review by ternaus — APPROVED** [review]

**Comment by ternaus** [comment]
Thank you for the PR!
